test/config_parser_SUITE.erl

```diff
@@ -160,6 +174,79 @@ compare_unordered_lists(L1, L2, F) ->
 
 compare_ordered_lists([H1|T1], [H1|T2], F) ->
     compare_ordered_lists(T1, T2, F);
+compare_ordered_lists([{backends, Backends}], [{backends, Backends2}], _) ->
+    Sns = proplists:get_value(sns, Backends, []),
+    Sns2 = proplists:get_value(sns, Backends2, []),
+    Push = proplists:get_value(push, Backends, []),
+    Push2 = proplists:get_value(push, Backends2, []),
+    Http = proplists:get_value(http, Backends, []),
+    Http2 = proplists:get_value(http, Backends2, []),
+    compare_unordered_lists(Sns, Sns2),
+    compare_unordered_lists(Push, Push2),
+    compare_unordered_lists(Http, Http2);
+compare_ordered_lists([{mod_event_pusher_rabbit, Opts}|T1], [{mod_event_pusher_rabbit, Opts2}|T2], F) ->
+    Presence = proplists:get_value(presence_exchange, Opts, []),
+    Presence2 = proplists:get_value(presence_exchange, Opts2, []),
+    Chat = proplists:get_value(chat_msg_exchange, Opts, []),
+    Chat2 = proplists:get_value(chat_msg_exchange, Opts2, []),
+    Group = proplists:get_value(groupchat_msg_exchange, Opts, []),
+    Group2 = proplists:get_value(groupchat_msg_exchange, Opts2, []),
+    compare_unordered_lists(Presence, Presence2),
+    compare_unordered_lists(Chat, Chat2),
+    compare_unordered_lists(Group, Group2),
+    compare_ordered_lists(T1, T2, F);
+compare_ordered_lists([{mod_keystore, Opts}|T1], [{mod_keystore, Opts2}|T2], F) ->
+    Keys1 = proplists:get_value(keys, Opts, []),
+    Keys2 = proplists:get_value(keys, Opts2, []),
+    compare_unordered_lists(Keys1, Keys2),
+    compare_unordered_lists(lists:keydelete(keys, 1, Opts), lists:keydelete(keys, 1, Opts2)),
+    compare_ordered_lists(T1, T2, F);
+compare_ordered_lists([{Name, Opts} = H1|T1], [{Name, Opts2} = H2|T2], F) ->
+    Names = [
```

Comment:
This makes the function clause too long, this list of statically-chosen keys could be moved to a helper function returning that list, I find it more readable 🤔

Reply:
Is there something in erlang like `module attribute` in elixir?

Reply:
I agree see that this piece of code is somewhat difficult to grasp and extend according to my initial intentions (which are apparently not clear), I think I need to propose a rework here as it is me who started it all.